### PR TITLE
Add vulkanmemoryallocator dependency for Skia; roll Skia to 4230297004

### PR DESCRIPTION
**Pull request by bdero**
Fixes https://github.com/flutter/flutter/issues/102806.

Should unblock the roll once https://skia-review.googlesource.com/c/skia/+/535356 lands.

### CODE REVIEW

**Review by zanderso — COMMENTED** [review]
It looks like the license check will need to be taught to skip some of the new files

**Comment by bdero** [comment]
Ugh, local build on macos host_debug_unopt is successful for me.

**Comment by bdero** [comment]
The license script reordered a few things. My diff tool is smart enough to see it's not the entire `licenses_third_party`, but I think git gave up and thinks it's the whole file.

File: tools/licenses/lib/main.dart [hunk 1]
```diff
@@ -1964,6 +1982,8 @@ class _RepositoryRootThirdPartyDirectory extends _RepositoryGenericThirdPartyDir
       return _RepositoryPkgDirectory(this, entry);
     if (entry.name == 'vulkan')
       return _RepositoryVulkanDirectory(this, entry);
+    if (!entry.fullName.endsWith('externals/vulkanmemoryallocator'))
```
Comment (zanderso) [inline tools/licenses/lib/main.dart:1985]: Should this condition be the opposite? That is, if it *does* end with the string, then we *do* want to return `_RepositoryVulkanMemoryAllocatorDirectory`?

Comment (bdero) [reply tools/licenses/lib/main.dart:1985]: D'oh, yes

**Review by zanderso — APPROVED** [review]

**Pull request merged**